archlinuxcn/notepad---git/PKGBUILD

```diff
@@ -1,36 +1,46 @@
 # Maintainer: taotieren <admin@taotieren.com>
 
 pkgname=notepad---git
-pkgver=1.23.r122.gb8786b0
-pkgrel=1
-pkgdesc="Notepad-- 是一个简单的国产跨平台文本编辑器，是替换 Notepad++ 的一种选择。其内置强大的代码对比功能，让你丢掉付费的 Beyond Compare。"
+pkgver=2.12.r5.ga89e9c6
+pkgrel=2
+pkgdesc="Notepad-- 是使用C++编写的轻量级文本编辑器, 简称ndd, 可以支持Window/Mac/Linux操作系统平台。"
 arch=('x86_64')
 url="https://gitee.com/cxasm/notepad--"
-license=('GPL3')
+license=('GPL-3.0-or-later')
 provides=(${pkgname%-git} notepadplugin)
 conflicts=(${pkgname%-git})
 _qt=qt5
 depends=($_qt-base
-        $_qt-xmlpatterns
-        qscintilla-$_qt)
+    $_qt-xmlpatterns
+    qscintilla-$_qt)
 makedepends=(cmake
-            ninja
-            git
-            $_qt-tools)
-source=("${pkgname}::git+${url}.git#branch=cmake-dev")
+    ninja
+    git
+    $_qt-tools)
+source=("${pkgname}::git+${url}.git")
 sha256sums=('SKIP')
 
 pkgver() {
     cd "${srcdir}/${pkgname}/"
-    git describe --long --tags | sed 's/v//g;s/\([^-]*-g\)/r\1/;s/-/./g'
+    git describe --long --tags | sed 's/^v//g;s/\([^-]*-g\)/r\1/;s/-/./g'
+}
+
+prepare()
+{
+    git -C "${srcdir}/${pkgname}" clean -dfx
 }
 
 build() {
-    cd "${srcdir}/${pkgname}/"
+    cd "${srcdir}/${pkgname}"
```

Comment:
```
/build/notepad---git/src/notepad---git/src/qscint/src/qscilexerr.cpp:184:1: warning: control reaches end of non-void function [-Wreturn-type]
```
```
AutoUic: /build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.ui: Warning: action `menuCompare' not declared
```
```
/build/notepad---git/src/notepad---git/src/rcglobal.cpp: In function ‘void showFileInExplorer(QString)’:
/build/notepad---git/src/notepad---git/src/rcglobal.cpp:80:30: warning: ‘static bool QProcess::startDetached(const QString&)’ is deprecated: Use QProcess::startDetached(const QString &program, const QStringList &arguments) instead [-Wdeprecated-declarations]
   80 |         process.startDetached(cmd);
      |         ~~~~~~~~~~~~~~~~~~~~~^~~~~
In file included from /usr/include/qt/QtCore/QProcess:1,
                 from /build/notepad---git/src/notepad---git/src/rcglobal.cpp:5:
/usr/include/qt/QtCore/qprocess.h:280:17: note: declared here
  280 |     static bool startDetached(const QString &command);
      |                 ^~~~~~~~~~~~~
```
```
/build/notepad---git/src/notepad---git/src/scintillaeditview.cpp:1311:59: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 3 has type ‘quint32’ {aka ‘unsigned int’} [-Wformat=]
 1311 |                                 sprintf(lineString, "%08lld ", startLineNumOffset + i);
      |                                                      ~~~~~^    ~~~~~~~~~~~~~~~~~~~~~~
      |                                                           |                       |
      |                                                           long long int           quint32 {aka unsigned int}
      |                                                      %08d
/build/notepad---git/src/notepad---git/src/scintillaeditview.cpp:1333:59: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 3 has type ‘quint32’ {aka ‘unsigned int’} [-Wformat=]
 1333 |                                 sprintf(lineString, "%08lld ", startLineNumOffset + i);
      |                                                      ~~~~~^    ~~~~~~~~~~~~~~~~~~~~~~
      |                                                           |                       |
      |                                                           long long int           quint32 {aka unsigned int}
      |                                                      %08d
[299/301] Building CXX object CMakeFiles/notepad--.dir/src/cceditor/ccnotepad.cpp.o
/build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp: In member function ‘void CCNotePad::saveTabEdit(int)’:
/build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp:5662:55: warning: ‘bool QVariant::operator>=(const QVariant&) const’ is deprecated [-Wdeprecated-declarations]
 5662 |                 if (pEdit->property(Edit_File_New) >= 0)
      |                                                       ^
In file included from /usr/include/qt/QtGui/qevent.h:50,
                 from /usr/include/qt/QtGui/QCloseEvent:1,
                 from /build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.h:7,
                 from /build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp:2:
/usr/include/qt/QtCore/qvariant.h:473:31: note: declared here
  473 |     QT_DEPRECATED inline bool operator>=(const QVariant &v) const
      |                               ^~~~~~~~
/build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp: In member function ‘void CCNotePad::slot_actionSaveAsFile_toggle(bool)’:
/build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp:5829:55: warning: ‘bool QVariant::operator>=(const QVariant&) const’ is deprecated [-Wdeprecated-declarations]
 5829 |                 if (pEdit->property(Edit_File_New) >= 0)
      |                                                       ^
/usr/include/qt/QtCore/qvariant.h:473:31: note: declared here
  473 |     QT_DEPRECATED inline bool operator>=(const QVariant &v) const
      |                               ^~~~~~~~
/build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp: In member function ‘void CCNotePad::saveTempFile(ScintillaEditView*, int, QSettings&)’:
/build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp:6119:55: warning: ‘bool QVariant::operator>=(const QVariant&) const’ is deprecated [-Wdeprecated-declarations]
 6119 |                 if (pEdit->property(Edit_File_New) >= 0)
      |                                                       ^
/usr/include/qt/QtCore/qvariant.h:473:31: note: declared here
  473 |     QT_DEPRECATED inline bool operator>=(const QVariant &v) const
      |                               ^~~~~~~~
/build/notepad---git/src/notepad---git/src/cceditor/ccnotepad.cpp:6139:47: warning: ‘bool QVariant::operator>=(const QVariant&) const’ is deprecated [-Wdeprecated-declarations]
 6139 |         if (pEdit->property(Edit_File_New) >= 0)
      |                                               ^
/usr/include/qt/QtCore/qvariant.h:473:31: note: declared here
  473 |     QT_DEPRECATED inline bool operator>=(const QVariant &v) const
      |                               ^~~~~~~~
```
```
/usr/src/debug/notepad---git/notepad---git/src/qscint/scintilla/lexers/LexGCode.cpp:77:8: warning: type ‘struct OptionsTxt’ violates the C++ One Definition Rule [-Wodr]
```
```
/usr/src/debug/notepad---git/notepad---git/src/qscint/scintilla/lexers/LexMMIXAL.cpp:106:14: warning: ‘__builtin_memmove’ reading 100 bytes from a region of size 99 [-Wstringop-overread]
```

Reply:
此类问题有反馈过上游仓库，作者已不再维护 linux 发行版为由拒绝修复。后续会通过本地补丁更新。

Reply:
Since the upstream has stopped supporting the Linux platform, I would suggest not to build this package anymore, rather than relying on patches to fix it, especially since this is a VCS package.